x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/action/SamlInitiateSingleSignOnRequest.java

```diff
@@ -37,6 +39,9 @@ public ActionRequestValidationException validate() {
         if (Strings.isNullOrEmpty(spEntityId)) {
             validationException = addValidationError("entity_id is missing", validationException);
         }
+        if (Strings.isNullOrEmpty(assertionConsumerService)) {
```

Comment:
Because cloud is always going to send it. And otherwise you end up with an API that is inconsistent.
The only reason it's not always manadatory, everywhere is that it doesn't make sense for metadata. Registerined an SP, and then being told that you need to provide its ACS just to get the IDP metadata back doesn't make sense (and was a problem for tests).

Reply:
We discussed this in our sync